static/js/templates.js

```diff
@@ -2,22 +2,18 @@ var templates = (function () {
 
 var exports = {};
 
+var template_paths = ['', 'settings/', 'widgets/'];
+
 exports.render = function (name, arg) {
-    if (Handlebars.templates === undefined) {
-        throw new Error("Cannot find compiled templates!");
-    }
-    if (Handlebars.templates[name] === undefined) {
-        throw new Error("Cannot find a template with this name: " + name
-              + ". If you are developing a new feature, this likely "
-              + "means you need to add the file static/templates/"
-              + name + ".handlebars");
+    for (var i = 0; i < template_paths.length; i += 1) {
```

Comment:
We can do this or add the path to all the references to settings/widgets.

Reply:
Whoa. I didn’t realize webpack has special support for `require()` on computed strings like that, but yes, apparently it’s [smart enough](https://webpack.js.org/guides/dependency-management/#require-with-expression) to parse that computation, find all the `*.handlebars` files within `static/templates`, compile them, and bundle them into a map for runtime dispatch. Wouldn’t have been my first plan but I guess it works!

We don’t want to catch exceptions from rendering here, so the `(arg)` should be lifted outside the `try…catch`. And using `require.context` might help webpack generate slightly more efficient code:

```js
var template_context = require.context('../templates', true, /\.handlebars$/);
var template_paths = ['./', './settings/', './widgets/'];

exports.render = function (name, arg) {
    for (var i = 0; i < template_paths.length; i += 1) {
        var template;
        try {
            template = template_context(template_paths[i] + name + '.handlebars');
        } catch (_e) {
            continue;
        }
        return template(arg);
    }
    throw new Error('Cannot find template ' + name + '.handlebars anywhere '
        + 'under the static/templates/ folder.');
};
```

I think I’d be in favor of adding explicit paths for settings and widgets so as to reduce the amount of magic here.

Reply:
Yeah, the only reason we didn't have explicit paths for rendering `settings/foo.handlebars` was because the old system didn't support naming things that way.  So I think passing e.g. `settings/foo.handlebars` into `.render` is the right approach over doing a loop here.